source/dotnet/Library/AdaptiveCards.Templating/AdaptiveCardTemplate.cs

```diff
@@ -132,11 +167,15 @@ public string Expand(EvaluationContext context, Func<string, object> nullSubstit
         /// </example>
         /// <seealso cref="EvaluationContext"/>
         /// <returns>json as string</returns>
-        public string Expand(object rootData, Func<string, object> nullSubstitutionOption = null)
+        public string Expand(object rootData, out ArrayList<string> errorLog, Func<string, object> nullSubstitutionOption = null)
```

Comment:
Don't change existing API. 

Also the API is hard to use. 
You may add new api like

```
public <string, error> Expand2(object rootData, Func<string, object> nullSubstitutionOption = null)
```

Reply:
I used wrapper methods that kept the initial method headers so that the existing API is still usable. Would that be okay?